### PR TITLE
ZBUG-2817 : Upgraded ClamAV to 0.105.1

### PR DESCRIPTION
**Pull request by umagmrit**

### CODE REVIEW

File: thirdparty/clamav/zimbra-clamav/debian/control [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Source: zimbra-clamav
 Build-Depends: debhelper (>= 9), m4, dpkg-dev (>= 1.15.7), zimbra-openssl-dev (>= 1.1.1h-1zimbra8.7b4ZAPPEND), zimbra-libxml2-dev,
- libz-dev, zimbra-libmilter-dev, libncurses5-dev
+ libz-dev, zimbra-libmilter-dev, libncurses5-dev, libpcre2-dev, libjson-c-dev, libbz2-dev, check
```
Comment (silentsakky) [inline thirdparty/clamav/zimbra-clamav/debian/control:3]: is `check` a package name?

Comment (umagmrit) [reply thirdparty/clamav/zimbra-clamav/debian/control:3]: yes

Comment (silentsakky) [inline thirdparty/clamav/zimbra-clamav/debian/control:3]: don't we need to install rust toolchain for compilation?

Comment (umagmrit) [reply thirdparty/clamav/zimbra-clamav/debian/control:3]: yes, we need to install rust toolchain. But it is not provided by package. There is a script to install rust toolchain.

Comment (umagmrit) [reply thirdparty/clamav/zimbra-clamav/debian/control:3]: https://www.rust-lang.org/tools/install

Comment (silentsakky) [reply thirdparty/clamav/zimbra-clamav/debian/control:3]: can we update makefile to install rust first if not present, I would like to reduce package dependecies to be installed in system when compiling these packages, so we can port this easily to circleci

**Review by silentsakky — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Comment by stondino00** [comment]
They have released 105.1.

0.105.1
ClamAV 0.105.1 is a critical patch release with the following fixes:

Upgrade the vendored UnRAR library to version 6.1.7.

Fix issue building macOS universal binaries in some configurations.

Silence error message when the logical signature maximum functionality level is lower than the current functionality level.

Fix scan error when scanning files containing malformed images that cannot be loaded to calculate an image fuzzy hash.

Fix logical signature "Intermediates" feature.

Relax constraints on slightly malformed ZIP archives that contain overlapping file entries.

**Review by abhishekdeshmukh2209 — APPROVED** [review]

**Review by zimsuchitgupta — APPROVED** [review]

File: thirdparty/clamav/install_rust_toolchain.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#Script to install Rust toolchain. Starting with ClamAV v0.105, the Rust toolchain is required to compile ClamAV.
+is_rust_install=0
+is_rust_present(){
+	for cmd in cargo rustc rustup rust-gdb rust-lldb rustdoc rustfmt; do
+	  if ! command -v $cmd >/dev/null 2>&1; then
```
Comment (dasiyogesh) [inline thirdparty/clamav/install_rust_toolchain.sh:5]: indentation doesn't looks right

**Review by silentsakky — APPROVED** [review]

**Review by dasiyogesh — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Comment by stondino00** [comment]
https://blog.clamav.net/2022/10/new-packages-for-clamav-01037-01044.html

image

**Comment by umagmrit** [comment]
> https://blog.clamav.net/2022/10/new-packages-for-clamav-01037-01044.html
> 
> image

https://github.com/Zimbra/packages/pull/174

**Pull request merged**